packages/esbuild/lib/index.js

```diff
@@ -116,15 +117,15 @@ export function esbuild(options) {
           '\n'
         messages = file.messages
       } catch (error_) {
-        const cause = /** @type {VFileMessage | Error} */ (error_)
+        const cause = /** @type (VFileMessage | Error) */ (error_)
         const message =
-          'reason' in cause
-            ? cause
-            : new VFileMessage('Cannot process MDX file with esbuild', {
-                cause,
-                ruleId: 'process-error',
-                source: '@mdx-js/esbuild'
-              })
+          new VFileMessage(
+            `Cannot process MDX file with esbuild:\n  ${error_}`, {
+            cause,
+            place: ('reason' in cause && cause.place) || undefined,
```

Comment:
```suggestion
            place: 'reason' in cause ? cause.place : undefined,
```

I prefer ternaries over of conditionals, which can fall through